metadata_service/proxy/base_proxy.py

```diff
@@ -0,0 +1,6 @@
+class BaseProxy:
```

Comment:
We can make this an interface, enforcing other clients to implement all functions needed, with same parameters.